src/backend/NodeForest.js

```diff
@@ -17,17 +17,41 @@ class NodeForest {
         return null;
     }
 
-    static getNode(nodeId, nodeRootId) {
-        return NodeForest.getNodeFromRoot(NodeForest.nodeForest[nodeRootId], nodeId);
+    static getNode(nodeId) {
+        let newNode = NodeForest.allNodes[nodeId];
+        if (newNode === undefined) {
+            newNode = null;
+        }
+        return newNode;
+    }
+
+    static insertNode(newNode) {
+        NodeForest.allNodes[newNode.nodeId] = newNode;
     }
 
     static insertRootNode(newNode) {
         // To insert node into a tree, please use
         // Get function to get the node, then set the operand correctly.
+        NodeForest.insertNode(newNode);
         NodeForest.nodeForest[newNode.nodeId] = newNode;
         // console.log(`added node of type "${typeof newNode}" with ID ${newNode.nodeId}`);
     }
 
+    static insertIntoBinaryNodeLeft(newNode, parentNodeId) {
```

Comment:
I will create an issue to fix this in the future. It is not an issue right now since this is our design at the moment, but adding new methods for each node type is bad OOP